### DISCUSSION TITLE
How to use Include with auto-reference?

### DISCUSSION THREAD

**Discussion by faiverson** [prisma/prisma#3926] [Q&A] [answered]
I have this model:
```prisma
model Tournament {
  id            String                @id @default(uuid())
  name          String
  type          String
  current_round String                @default("1")
  matches       Match[]               @relation("matches_tournament_foreign")
}

model Match {
  id            String        @id @default(uuid())
  tournament_id String
  local_team_id String
  local_score   Int?
  away_team_id  String
  away_score    Int?
  round         String
```

Checking the docs using `includes` is possible if I have a variable:
```ts
const current_round = 1
prisma.tournament.findOne({
          where: {
            id
          },
          include: {
            matches: {
              where: {
                round: current_round
              },
            }
          }
        })
```

What I want is to filter the matches based on the current round number and the tournament ID.
But if I need to get the value referencing a model, not sure how to do it. 

Something like this:
``````
prisma.tournament.findOne({
          where: {
            id
          },
          include: {
            matches: {
              where: {
                round: tournament.round
              },
            }
          }
        })
```

In mysql you can do this with a subquery
Is this possible or I need to make a query first to get the `round`?

**Comment by ryands17** [comment]
@faiverson 👋 
This should work:

```ts
const current_round = '1'
await prisma.tournament.findOne({
    where: {
      id: 'adadasdsd',
    },
    select: {
      matches: { where: { round: current_round }, select: { round: true } },
    },
})
```

**Reply by faiverson** [reply]
But the problem is that I don't want to set a variable for `current_round`, what I want is to point current_round from the model.
Something more like this:
```
prisma.tournament.findOne({
          where: {
            id
          },
          include: {
            matches: {
              where: {
                round: tournament.round // if I replace this with a variable works, but I want to get this from the model Tournament
              },
            }
          }
        })
```

**Reply by ryands17** [reply, marked as answer]
Unfortunately that's not possible at the moment. I would suggest opening a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into this :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
